src/main/java/seedu/address/model/DateTime.java

```diff
@@ -0,0 +1,61 @@
+package seedu.address.model;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.time.DateTimeException;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+
+/**
+ * Represents a date or time (optional) within PalPay.
+ * Guarantees:  immutable; is valid as declared in {@link #isValidDate(String)}
+ */
+public class DateTime {
+
+    public static final String MESSAGE_CONSTRAINTS = "DateTime objects must adhere to the format: DDMMYYYY [HH:mm]";
+
+    public final String value;
+
+    public static final String VALIDATION_REGEX_DATE = "(0?[1-9]|[12][0-9]|3[01])(0?[1-9]|1[0-2])\\d{4}";
```

Comment:
I don't think this regex is correct leh. 

`\\d{4}` means `\dddd` literally.

Reply:
For generality's sake I didn't make it compulsory for dateTime to be futuristic (in case we use it in our ```Transaction``` history too). Thus, it might be better to do future/past date validation in the individual classes instead, as per their requirement (e.g. past for view commands and future for project commands)

Reply:
What's an example of a date that will pass isValidDate?

Reply:
any value from ```01010000 00:00``` to ```12129999 23:59```

Reply:
although, hang on this for a while, I'm trying to see if using a DateTimeFormatter + LocalDate and Optional<LocalTime> is appropriate

Reply:
Yeah I meant the `\\d{4}` literally means `\dddd` and not 4 digits like `0000`

Reply:
Oh my bad. It's my mistake. The  `\` is to escape the next `\` for Java.